sdks/python/apache_beam/runners/interactive/interactive_environment_test.py

```diff
@@ -88,6 +91,72 @@ def test_watch_class_instance(self):
     self.assertVariableWatched('_var_in_class_instance',
                                self._var_in_class_instance)
 
+  def test_fail_to_set_pipeline_result_key_not_pipeline(self):
+    class NotPipeline(object):
+      pass
+
+    with self.assertRaises(AssertionError) as ctx:
+      ie.current_env().set_pipeline_result(NotPipeline(),
+                                           runner.PipelineResult(
+                                               runner.PipelineState.RUNNING))
+      self.assertTrue('pipeline must be an instance of apache_beam.Pipeline '
+                      'or its subclass' in ctx.exception)
+
+  def test_fail_to_set_pipeline_result_value_not_pipeline_result(self):
+    class NotResult(object):
+      pass
+
+    with self.assertRaises(AssertionError) as ctx:
+      ie.current_env().set_pipeline_result(self._p, NotResult())
+      self.assertTrue('result must be an instance of '
+                      'apache_beam.runners.runner.PipelineResult or its '
+                      'subclass' in ctx.exception)
+
+  def test_set_pipeline_result_successfully(self):
+    class PipelineSubClass(beam.Pipeline):
+      pass
+
+    class PipelineResultSubClass(runner.PipelineResult):
+      pass
+
+    pipeline = PipelineSubClass()
+    pipeline_result = PipelineResultSubClass(runner.PipelineState.RUNNING)
+    ie.current_env().set_pipeline_result(pipeline, pipeline_result)
+    self.assertIs(ie.current_env().pipeline_result(pipeline), pipeline_result)
+
+  def test_determine_terminal_state(self):
+    for state in (runner.PipelineState.DONE,
+                  runner.PipelineState.FAILED,
+                  runner.PipelineState.CANCELLED,
+                  runner.PipelineState.UPDATED,
+                  runner.PipelineState.DRAINED):
+      ie.current_env().set_pipeline_result(self._p, runner.PipelineResult(
```

Comment:
PipelineState has an is_terminal method defined on the class that can be used here.

Reply:
Great, thanks! I totally missed it. I'll change it to use that class method.